content/articles/build-a-twitch-chatbot-in-nodejs/index.md

```diff
@@ -0,0 +1,436 @@
+// Any error found shall be logged out in the console
+client.connect().catch(console.error);
+
+client.on('message', (channel, tags, message, self) => {
+    // Lack of this statement or it's inverse (!self) will make it in active
+    if (self) return;
+
+    // Create up a switch statement with some possible commands and their outputs
+    // The input shall be converted to lowercase form first
+    // The outputs shall be in the chats
+    switch (message.toLowerCase()) {
+        // Use tags to obtain the username of the one who has keyed in a certain input
+        case 'commands':
+            client.say(channel, `@${tags.username}, available commands are:
+            Commands Help Greetings Hi !Website !Name
+           
+            For more help just type "Help"
+            `);
+            break;
+        case '!website':
+            client.say(channel, `@${tags.username}, my website is www.section.io!`);
+            break;
+        case 'greetings':
+            client.say(channel, `Hello @${tags.username}, what's up?!`);
+            break;
+        case 'hi':
+            client.say(channel, `${tags.username}, hola!`);
+            break;
+        case '!name':
+            client.say(channel, `Hello @${tags.username}, my name is ChatBot! Type "help" to continue...`);
+            break;
+        case 'help':
+            client.say(channel, `${tags.username}, Use the following commands to get quick help:
+            -> Commands: Get Commands || 
+            Help: Get Help || 
+            Greetings: Get Greetings || 
+            Hi: Get "Hola" || 
+            !Website: Get my website || 
+            !Name: Get my name || 
+            !Upvote first_name second_name: Upvote user first_name second_name ||  Upvote first_name second_name: Upvote user first_name second_name || 
+            !Cheer first_name second_name: Cheer first_name second_name || Cheers first_name second_name: Cheer first_name second_name --
+
+            For more help just ping me up!
+            `);
+            break;
+        default:
+
+            // We shall convert the message into a string in which we shall check for its first word
+            // and use the others for output
+            let mymessage = message.toString();
+            if ((mymessage.split(' ')[0]).toLowerCase() === '!upvote' || 'upvote') {
+                // You can add some emojis which will appear in the chat using their emoji names
+                // For example "PopCorn"
+                client.say(channel, `TwitchLit @${(mymessage.split(' ')[1] + '_' + mymessage.split(' ')[2])} TwitchLit  you have been UPVOTED by ${ tags.username }`);
+
+            } else if ((mymessage.split(' ')[0]).toLowerCase() === '!cheer' || 'cheers') {
+                console.log(`HSCheers @${(mymessage.split(' ')[1] + '_' + mymessage.split(' ')[2])} HSCheers you have been UPVOTED by ${ tags.username }`);
+            }
+            break;
+    }
+});
+```
```

Comment:
Break down this code snippet into smaller code snippets, and explain each of them in-detail.

Reply:
Ok

Reply:
Done!

Reply:
Committed the suggestions